test/e2e/pages/loginPage.ts

```diff
@@ -20,29 +20,83 @@ class LoginPage extends Page {
-        const nextButtonElement = element(by.xpath('//XCUIElementTypeButton[@label="Next"]'));
-        nextButtonElement.click();
+        const microsoftOnlineContinue = element(by.xpath(`//XCUIElementTypeButton[@name="Continue"]`));
+        browser.wait(ExpectedConditions.presenceOf(microsoftOnlineContinue));
+        microsoftOnlineContinue.click().then((result) => {
+          return browser.sleep(TEST_CONFIG.PAGE_LOAD_WAIT);
+        })
+          .then((result) => {
+            const useAnotherAccountButton = element(by.xpath(`//XCUIElementTypeOther[@name="Use another account, Use another account"]`));
+            useAnotherAccountButton.isPresent().then((result) => {
+              if (result) {
+                return useAnotherAccountButton.click();
+              }
+              else {
+                return ;
+              }
+            });
+
+          })
+          .then((result) => {
+            // Fill in username and click Next
+            const usernameFld = this.getUsernameField();
+            browser.wait(ExpectedConditions.presenceOf(usernameFld));
+            usernameFld.sendKeys(username);
+
+            const nextButtonElement = element(by.xpath('(//XCUIElementTypeButton[@name="Next"])[1]'));
+            nextButtonElement.click();
+
+            const passwordForOrgAccount = element(by.xpath(`//XCUIElementTypeOther[@name="Sign In"]/XCUIElementTypeOther[4]/XCUIElementTypeSecureTextField`));
+            browser.wait(ExpectedConditions.presenceOf(passwordForOrgAccount));
+            passwordForOrgAccount.sendKeys(password);
+
+            const signinForOrgAccount = element(by.xpath(`//XCUIElementTypeButton[@name="Sign in"]`));
+            return signinForOrgAccount.click();
+          })
+          .then((result) => {
+            return browser.sleep(TEST_CONFIG.PAGE_LOAD_WAIT);
+          })
+          .then((result) => {
+            const authCContinue = element(by.xpath(`//XCUIElementTypeButton[@name="Continue"]`));
+            return authCContinue.isPresent()
+              .then((result) => {
+                if (result) {
+                  return authCContinue.click();
+                }
+                else {
+                  const acceptButton =   element(by.xpath(`//XCUIElementTypeButton[@name="Accept"]`));
+                  return acceptButton.click();
+                }
+              });
+          });
+
+        //
+
+
+
 
         // Fill in password and click Sign in
-        const pFld = element(by.xpath(`//XCUIElementTypeOther[@name="Sign In"]/XCUIElementTypeOther[4]/XCUIElementTypeSecureTextField`));
-        browser.wait(ExpectedConditions.presenceOf(pFld));
-        pFld.sendKeys(password);
-        const signInButtonElement = element(by.xpath('//XCUIElementTypeButton[@name="Sign in"]'));
-        signInButtonElement.click();
+        // const pFld = element(by.xpath(`//XCUIElementTypeSecureTextField[@label="Enter the password for ${username}"]`));
```

Comment:
why the commented out code in the code base?

Reply:
done